backend/src/main/java/gov/cdc/usds/simplereport/db/repository/SpecimenTypeRepository.java

```diff
@@ -0,0 +1,11 @@
+package gov.cdc.usds.simplereport.db.repository;
+
+import java.util.Optional;
+
+import gov.cdc.usds.simplereport.db.model.SpecimenType;
+
+public interface SpecimenTypeRepository extends EternalEntityRepository<SpecimenType> {
+
+    Optional<SpecimenType> findByTypeCode(String swabType);
```

Comment:
Hmm. Probably so, yes. The backward-compatibility shims get weird, but I think the answer is "in the strictly implausible event that somebody soft-deletes a SpecimenType, that shim should fail.

Reply:
Ah, no it shouldn't: if this doesn't return an object, we'll create one (in the `createDeviceType` service call). So, have to flag that in comments somehow, because by the repository contract it clearly should. (And in fact, I started to write the SpecimenType service and simply assumed that it did check, despite the fact that I wrote it myself and knew it didn't.) Probably should be marked `@Deprecated` even though that has potentially annoying linting implications.

Reply:
i'm not sure this is the right approach. the way it is now, we're allowing `DeviceSpecimen`s to be saved downstream (in `createDeviceType`) with `SpecimenType`s that have already been deleted...? at the very least, if we are going to retrieve deleted `SpecimenTypes` that we then save as FKs in other non-deleted DB records, we should un-delete the `SpecimenType` before we save it in a new `DeviceSpecimen` elsewhere.

Reply:
The current API does not have a method for deleting specimen types, and the purpose of this method is to maintain backward compatibility until the final API is rolled out—I've marked it deprecated to make it clear why it's there.